### ISSUE TITLE
Security layer?

### ISSUE THREAD

**Issue by lawlsausage** [stvcott8511/monitor#4]
Will monitor support authenticated access? What about multi-tenant support in the vein of Slack type "teams"?

**Comment by stvcott8511** [comment]
Yes, API should and will need to be authenticated. I could use some of your thoughts on it @lawlsausage

**Comment by lawlsausage** [comment]
For the UI, implementing JWT style authentication tokens would work fine with single way hashing of passwords saved to a database table for basic support.

As for external applications, the API should implement OAuth2 and make use of providing client keys and secrets.
